docs/articles/documentation/extending-testcafe/custom-browser-provider-plugin/browser-provider-methods.md

```diff
@@ -0,0 +1,127 @@
+```
+
+Parameter     | Type   | Description
+------------- | ------ | ----------------------------------------------------------------------------------------------------------------
+`id`          | String | A unique identifier generated by TestCafe that you can use to identify a browser window through the life cycle. The identifier is included to the window title exposed via the `document.title` property.
+`pageUrl`     | String | The url of the page to open.
+`browserName` | String | The browser name.
+
+See [Example](index.md#example).
+
+### closeBrowser
+
+Closes the specified browser.
+
+```text
+async closeBrowser(id)
+```
+
+Parameter   | Type   | Description
+----------- | ------ | ---------------------------------------------------------------------------------------------------------------
+`id`        | String | A unique identifier generated by TestCafe that you can use to identify a browser window through the life cycle.
+
+See [Example](index.md#example).
+
+## Optional Methods
+
+### init
+
+Performs initialization actions the provider may require. The method is called in the `testcafe` process life cycle once, before other provider methods are called.
+
+```text
+async init()
+```
+
+For example, the existing [PhantomJS browser provider](https://github.com/DevExpress/testcafe-browser-provider-phantomjs) uses this method to create a PhantomJS instance.
+
+### dispose
+
+Performs cleanup actions for the provider data. The method is called before the `testcafe` process is finished.
+
+```text
+async dispose()
+```
+
+For example, the existing [PhantomJS browser provider](https://github.com/DevExpress/testcafe-browser-provider-phantomjs) uses this method to destroy a PhantomJS instance.
+
+### getBrowserList
+
+Returns the array of names used to identify the browsers. This method is used if the provider supports multiple browsers.
+
+```text
+async getBrowserList()
+```
+
+See [Example](index.md#example).
+
+### isValidBrowserName
+
+Checks if the specified browser name exists. This method is used if the provider supports multiple browsers.
+
+```text
+async isValidBrowserName(browserName)
+```
+
+Parameter     | Type   | Description
+------------- | ------ | -------------------------------
+`browserName` | String | The browser name to validate.
+
+Returns `true` if the specified name exists, or `false` otherwise.
+
+See [Example](index.md#example).
+
+### resizeWindow
+
+Resizes the browser window to the specified width and height.
+
+```text
+async resizeWindow(id, width, height, currentWidth, currentHeight)
+```
+
+Parameter   | Type    | Description
+----------- | ------- | ----------------------------------------------------------------------------------------------------------------
+`id`        | String  | A unique identifier generated by TestCafe that you can use to identify a browser window through the life cycle.
+`width`     | Number  | The width to resize the browser window to, in pixels.
```

Comment:
Not the browser window, but the browser window client area (only the part of the window that displays a web page, without menus, status bars, tabs, etc.)